### PR TITLE
DON'T USE ALL CAPS for module types

### PR DESCRIPTION
**Pull request by samoht**
All module types now use the normal Snake case.

(this will please @mshinwell)

### CODE REVIEW

File: src/irmin-mirage/git/irmin_mirage_git.mli [hunk 1]
```diff
@@ -21,25 +21,25 @@ module type S_MAKER = functor
...
+module type KV_maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
   S
     with type key = string list
      and type step = string
      and type contents = C.t
      and type branch = string
      and module Git = G
 
-module type REF_MAKER = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
+module type REF_Maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
```
Comment (CraigFe) [inline src/irmin-mirage/git/irmin_mirage_git.mli:32]:
```suggestion
module type Ref_maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
```

File: src/irmin-pack/layered/s.ml [hunk 2]
```diff
@@ -32,10 +32,10 @@ module type LAYERED = sig
   include LAYERED_GENERAL with type _ t := t
 end
 
-module type LAYERED_ATOMIC_WRITE_STORE = sig
-  include ATOMIC_WRITE_STORE
-  module U : ATOMIC_WRITE_STORE
-  module L : ATOMIC_WRITE_STORE
+module type LAYERED_Atomic_write_store = sig
```
Comment (CraigFe) [inline src/irmin-pack/layered/s.ml:35]:
```suggestion
module type Layered_atomic_write_store = sig
```

And likewise for the other types here.

File: src/irmin-unix/xgit.mli [hunk 3]
```diff
@@ -37,25 +37,25 @@ module type S_MAKER = functor
...
+module type KV_maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
   S
     with type key = string list
      and type step = string
      and type contents = C.t
      and type branch = string
      and module Git = G
 
-module type REF_MAKER = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
+module type REF_Maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
```
Comment (CraigFe) [inline src/irmin-unix/xgit.mli:48]:
```suggestion
module type Ref_maker = functor (G : Irmin_git.G) (C : Irmin.Contents.S) ->
```

File: src/irmin/irmin.mli [hunk 4]
```diff
@@ -30,7 +30,7 @@
     Consult the {!basics} and {!examples} of use for a quick start. See also the
     {{!Irmin_unix} documentation} for the unix backends.
 
-    {e Release %%VERSION%% - %%HOMEPAGE%%} *)
+    {e Release %%Version%% - %%HOMEPAGE%%} *)
```
Comment (CraigFe) [inline src/irmin/irmin.mli:33]: Oops.
```suggestion
    {e Release %%VERSION%% - %%HOMEPAGE%%} *)
```

File: src/irmin/version.ml [hunk 5]
```diff
@@ -1 +1 @@
-let current = "%%VERSION%%"
+let current = "%%Version%%"
```
Comment (CraigFe) [inline src/irmin/version.ml:1]:
```suggestion
let current = "%%VERSION%%"
```

**Review by CraigFe — COMMENTED** [review]
Much less shouty :-)

**Comment by Ngoguey42** [comment]
Oh, I like the shouty case!

**Comment by mattiasdrp** [comment]
This looks nice! My only concern is that ALL CAPS allowed to know what were module types and what were not. Should we agree on a naming convention for module types or just "know" they're module types?

**Comment by samoht** [comment]
@mattiasdrp there are no syntactic distinction between types and values, why should we add a distinction between modules and modules types? :-) More seriously, I agree that consistency is a good goal - my next set of patches will remove s.ml and move all the store modules and module types in their own modules, with similar name conventions.

**Comment by Ngoguey42** [comment]
I'll pay you a 🍺 !

**Comment by samoht** [comment]
> I'll pay you a 🍺 !

Only one? That's clearly not enough :-) MERGIIIIING

**Pull request merged**